src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/CustomItemEffects.java

```diff
@@ -533,6 +567,44 @@ public int countItemsInInventoryAndStorage(ItemStack match) {
 		return count;
 	}
 
+	public ItemStack getFirstItemInRuler() {
+		int count = 0;
+
+		ItemStack held = Minecraft.getMinecraft().thePlayer.getHeldItem();
+		String heldInternal = NotEnoughUpdates.INSTANCE.manager.getInternalNameForItem(held);
+
+		if (heldInternal != null && heldInternal.equals("BUILDERS_RULER")) {
```

Comment:
Consider using an early return style to avoid that many indentation levels (which make most of your code utterly unreadable).

```java
	if (condition) return;
	// Rest of code
```
instead of 
```java
	if (!condition) {
		// Rest of code
	}
```
generally you should put the negative response earlier (since those tend to be shorter). this is especially compounding when you repeat this bad if style about a thousand times instead of merging the if conditions together. someone really seems to be having a fetish for a thousand tabs before each line of code